src/app/app.config.ts

```diff
@@ -1,12 +1,19 @@
-import { ApplicationConfig } from '@angular/core';
+import { ApplicationConfig, isDevMode } from '@angular/core';
 import { provideRouter, withViewTransitions } from '@angular/router';
 
 import { routes } from './app.routes';
 import { provideClientHydration } from '@angular/platform-browser';
+import { provideStore } from '@ngrx/store';
+import { provideEffects } from '@ngrx/effects';
+import { provideStoreDevtools } from '@ngrx/store-devtools';
+import { ROOT_REDUCERS, AllEffects } from './store/app.state';
 
 export const appConfig: ApplicationConfig = {
   providers: [
     provideRouter(routes, withViewTransitions()),
     provideClientHydration(),
-  ],
+    provideStore(ROOT_REDUCERS),
+    provideEffects(AllEffects),
+    provideStoreDevtools({ maxAge: 25, logOnly: !isDevMode() })
```

Comment:
maxAge: 25, logOnly: !isDevMode(): I have a question, what is this line for?

Reply:
It is the base configuration of the schematics of ngrx devtools, it is used to be able to debug the state of the store on the web.
maxAge : saves the last states in this case lists the last 25 changes
logOnly: Restricts the app to development mode only